corehq/apps/accounting/models.py

```diff
@@ -2374,6 +2510,157 @@ class Meta:
+        customer = stripe.Customer.create(
+            description="{}'s cards".format(self.web_user),
+            email=self.web_user,
+        )
+        self.customer_id = customer.id
+        self.save()
+        return customer
+
+    def _get_stripe_customer(self):
+        return stripe.Customer.retrieve(self.customer_id)
+
+    @property
+    def all_cards(self):
+        return self.customer.cards.data
+
+    def all_cards_serialized(self, billing_account):
+        return [{
+            'brand': card.brand,
+            'last4': card.last4,
+            'exp_month': card.exp_month,
+            'exp_year': card.exp_year,
+            'token': card.id,
+            'is_autopay': card.metadata.get('auto_pay_{}'.format(billing_account.id), False),
+        } for card in self.all_cards]
+
+    def get_card(self, card_token):
+        return self.customer.cards.retrieve(card_token)
+
+    def get_autopay_card(self, billing_account):
+        return next((card for card in self.all_cards
+                     if card.metadata.get(self._auto_pay_card_metadata_key(billing_account)) == 'True'),
+                    None)
+
+    def remove_card(self, card_token):
+        card = self.get_card(card_token)
+        self._remove_card_from_all_accounts(card)
+        card.delete()
+
+    def _remove_card_from_all_accounts(self, card):
+        accounts = BillingAccount.objects.filter(auto_pay_user=self.web_user)
+        for account in accounts:
+            if account.autopay_card == card:
+                account.remove_autopay_user()
+
+    def create_card(self, stripe_token, billing_account, autopay=False):
+        customer = self.customer
+        card = customer.cards.create(card=stripe_token)
+        self.set_default_card(card)
+        if autopay:
+            self.set_autopay(card, billing_account)
+        return card
+
+    def set_default_card(self, card, save=True):
+        self.customer.default_card = card
+        if save:
+            self.customer.save()
+        return card
+
+    def set_autopay(self, card, billing_account):
+        """
+        Sets the auto_pay status on the card for a billing account
+
+        If there are other cards that auto_pay for that billing account, remove them
+        """
+        if billing_account.auto_pay_enabled:
+            self._remove_other_auto_pay_cards(billing_account)
+
+        self._update_autopay_status(card, billing_account, autopay=True)
+        billing_account.update_autopay_user(self.web_user)
+
+    def unset_autopay(self, card, billing_account):
+        """
+        Unsets the auto_pay status for this card, and removes it from the billing account
+        """
+        if card.metadata[self._auto_pay_card_metadata_key(billing_account)] == "True":
+            self._update_autopay_status(card, billing_account, autopay=False)
+            billing_account.remove_autopay_user()
+
+    def _update_autopay_status(self, card, billing_account, autopay, save=True):
```

Comment:
it seems like all the invocations of this function default to `save=True` maybe we could get rid of this param?